### ISSUE TITLE
Priorities and Sorting

### ISSUE THREAD

**Issue by cj1601** [AppliedEnergistics/Applied-Energistics-2#1655]
In AE1 the behaviour of items going into the network was that it would first search for a cell that already contains the item, if it found one it would place it in that cell even if there was a cell with higher priority somewhere else in the network. With the new behaviour an item will go into a higher priority cell even if it is empty and there is already a cell with the item in question at a lower priority.

I don’t like using drives for my main storage because the items get randomly put onto the cells. I like to use chests so that I can manually sort the items going into the network by placing them into the cells I chose. Using a partition does not fulfill this purpose either because I would have to reformat the cell every time I put a new item into it.

In AE1 I could have a high priority cell that any overflow I put into the system would go to. I could then decide what cells I wanted that overflow to go into manually. As it currently stands in AE2 I cannot do this as having a cell with a higher priority will have everything go into it, as appose to just items that didn’t already have a place to go as it was in AE1.

It would be nice to a) restore the old functionally where it will first search for a cell that already contains the item or b) add a card that would change the behaviour so that we can have both the new style and the old style

**Comment by yueh** [comment]
Not possible as card, the sorting is way too deep to be replace by something simple like a card. Mostly the same for a config option.

I currently do not see any particular advantage over the current system for it. Something like a overflow chest is still possible, but probably requires a different approach.

It will mostly be a inferior solution with some disadvantages, which are then no longer be possible to be solved with a different design approach.

For example it would no longer be possible to define a new location for an already existing item and use this without completely emptying the network of this item, worst case even cycle the complete network inventory, because there is still a cell somewhere with a single stack, which will be used as target before the one with an actually higher priority.

Also the network is self cleaning itself in this cases, as it will use the items in inventories with a lower priority for exporting or crafting before the higher ones. If the items are actually used it will just slowly fix itself or you can use a simple loop and export these items. With AE1 you would have to build a temporary storage, export everything into it and then only when not a single item is left inside the network the same thing but in reverse. If the input would be faster (say a quarry) then you could export it, it would even be impossible to use the new storage without stopping the input until done.

To summarise it, it will mostly require that you trade the effort to design your network based on your needs once and let the network then handle the maintenance with a slightly easier network design, but then you have to do the maintenance by yourself, which requires way more effort when done more than for 1 or 2 items. Maybe even require to shutdown half your base until it is done.
And then one of the important intentions of AE2 is to design your network and not to constantly do a manual maintenance.

**Comment by cj1601** [comment]
I just want to make sure we are on the same page then ask for some clarification. In AE 1 all I had to do was build the higher priority overflow chest. I like to not have to use interfaces if I'm in my main storage area. In this way I can take things and put them in and out of ME Chests manually, thus choosing what goes onto those cells without having to whitelist or blacklist things every time a new item is introduced (usually more often then not). Once I have this setup (the sorting is usually already done in normal chests as it takes some time to gather enough materials to build the AE network, I then just have to replace the normal chests with ME Chests(or use a storage bus if I think the chest is of sufficient size). Now add the high priority storage cell, done, sorted system with overflow control.

I may be missing something, but short of reformatting cells every time I want to add a new item I've yet to see a way to do this in AE2. I would not mind building a bit of a system that would accomplish this but I'm at a loss as to how it would work. If I'm missing something let me know =), I'll put my closing notes for this post bellow to sort out everything.

What I want to accomplish is as follows:
a)I have any number of ME Chests in my main storage area where I can access the items in and out without having to go through a ME Interface to do so.
b)When I do want to use an interface any items (left over from crafting or me putting them in the system) will go to one of the above mention chests if one of the items already exists in them first.
c)There is then a ME Chest where any item that does not already have a cell to go into will end up, thus allowing me to sort these items into the chests that I chose.
d)To my knowledge this cannot be done with a partition because I'd have to edit the partition every time a new unique item is introduced to the system (again this is actually quite often)

As a final clarification, the basics of what I am saying is that I could easily do this in AE1 with an "overflow" chest that had a higher priority. That method does not work in AE2 as any chest with a higher priority will collect everything you put into an interface or is left over from auto crafting. If I'm missing something and there is a way to do this let me know, if not I'd like to purpose similar functionality as a future change. From my perspective, it seems as though a feature was removed between AE1 and AE2 although I know that it is just new behaviour in the way things are sorted.

Thanks =),
CJ

**Comment by yueh** [comment]
I think the main issue here is, that AE is really designed to centralise all inventories and make it accessible through a single location (the terminals). 

Trying to avoid it and use it as a filter system is not really something which AE2 is designed for.
Also there are enough other mods, which really fit better for this use case (XU, BC, LP, etc. basically any real pipe mod) and I do not think we can really to cover every possible use case. Basically use the right tool for the job.

Of course there are ways to do it with AE2, but these are a bit more clunky.
For example storage buses have a button to configure the whitelist to insert items based on their attached inventory contents. So you could just use normal chests, dump the item in there and click the button to refresh the whitelist.

**Comment by cj1601** [comment]
"I think the main issue here is, that AE is really designed to centralise all inventories and make it accessible through a single location (the terminals). "

I know and I just want to be 100% perfectly clear, as I said I feel as though a feature has been removed between AE1 and AE2 because I was able to do this with no difficulty in AE1. I would still propose this "feature" be added back in some capacity. I think AE is a brilliant mod and I like the direction it has gone but the mechanics changes are mechanics changes(and make for a very interesting experience compared to AE1), again, it just feels like something had been completely removed with the "problem" with the current priority system. And I do want to also make perfectly clear that I understand the stance and that it is not a "bug" but new functionality of how items get sorted into cells altogether.

This last bit should probably be saved for a private forum or an other, but what "clunky" methods would be involved in making the system behave in the way I above described. I'm not trying to shy away from making a well built system, I'd just like my system to behave the way it used to in AE1, I do not mind working a little bit to achieve this goal. I guess I just don't yet understand the AE2 mechanis well enough to be able to build a system that would mimic the old AE1 behavior well enough yet to build this "clunky" system you speak of.

As said above, I'd still like to keep this as a "feature request" or something along those lines, but in the mean time any insight as to how I might be able to achieve my intended goal (without the use of other mods, I did mention that I like AE right?) would be great.

Thanks,
CJ

**Comment by yueh** [comment]
After some more reconsideration, I do not think this possible with a reasonable amount of work.
It is a huge change and not something which can be resolved with a config option or some upgrade card.

We can probably reconsider it, when more requests for it appear. But as said it will be a breaking change and and the tradeoff will be, that some builds will no longer be possible. For having a single use case being a bit easier.

**Comment by Blue64** [comment]
This is possible without reworking the system, you just need to alter the priorities. You set the overflow to Sub-High and whitelist partitions set to High. You'd not need to partition every time, just partition once and it can all go through that storage whitelist first, then it gets sent to the next priority down which is the overflow chest, so on and so forth. You don't need to reformat the entire system for every new unique item (dyes, cobble, quartz, etc.) Because it's not checking a blacklist, it's checking a whitelist, this also allows you to set the priority as high as you'd like for that drive.

**Comment by Blue64** [comment]
Yueh, I actually consider the whitelist to be the streamline, because you control precisely what goes through, it is the ultimate control, ultimate precision, and you can set it to be as relaxed or strict as you'd like. I don't know why you'd consider it clunky.
